lib/yocaml/runtime.ml

```diff
@@ -1,84 +1,103 @@
 open Util
 
 module type RUNTIME = sig
-  val get_time : unit -> float
-  val file_exists : Filepath.t -> bool
-  val target_exists : Filepath.t -> bool
-  val is_directory : Filepath.t -> bool
-  val get_modification_time : Filepath.t -> int Try.t
-  val target_modification_time : Filepath.t -> int Try.t
-  val read_file : Filepath.t -> string Try.t
-  val content_changes : Filepath.t -> string -> bool Try.t
-  val write_file : Filepath.t -> string -> unit Try.t
-  val read_dir : Filepath.t -> Filepath.t list
-  val create_dir : ?file_perm:int -> Filepath.t -> unit
-  val log : Log.level -> string -> unit
+  type 'a t
+
+  val bind : 'a t -> ('a -> 'b t) -> 'b t
+  val return : 'a -> 'a t
```

Comment:
I think waiting explicitly for a `Preface.Specs.Monad.CORE` (or `Preface.Specs.MONAD`, but i'm not sure that `Lwt`, for example, fit with a monad in a preface sense) module would be better. It would be assumed that the runtime can be built in this way: 

```ocaml
module My_runtime = struct 
   module Monad = Preface.Make.Monad.Via_return_and_bind (Lwt)
   let get_time = ...
end
```

Reply:
I think @dinosaure's approach is the right one. 
Since only `bind` and `return` are required, I'm not sure that building a monad on top of an already existing monad (in this case, Lwt) is useful and relevant. And besides it avoids burying the monadic operators in a submodule.

Reply:
Yes, I resolve the conversation.